File: 253MeetingRoomsII.py


def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        start, end = [], []
        
        s = e = 0
        rooms, available =0, 0
        
        for i in range(len(intervals)):
            start.append(intervals[i][0])
            end.append(intervals[i][1])
            
        start.sort()
        end.sort()
        
        while s < len(start):
            if start[s] < end[e]:
                if available > 0:
                    available -= 1
                    
                else:
                    rooms += 1
                s += 1
            else:
                available += 1
                e += 1
        return rooms

File: 253.MeetingRoomsII.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        int[] start = new int[intervals.length];
        int[] end = new int[intervals.length];

        for(int i = 0; i < intervals.length; i++){
            start[i] = intervals[i][0];
            end[i] = intervals[i][1];
        }

        Arrays.sort(start);
        Arrays.sort(end);

        int s = 0;
        int e = 0;
        int count = 0;
        int res = 0;

        while (s < intervals.length) {
            if (start[s] < end[e]){
                count++;
                s++;
            } else{
                count--;
                e++;
            }
            res = Math.max(res, count);
        }


        return res;
    }
}
